packages/gestalt/src/Upsell.js

```diff
@@ -154,15 +157,21 @@ export default function Upsell({
             </Box>
           </Box>
         </Box>
-        <Box smDisplay="flex" marginStart="auto" smMarginEnd={4} smPaddingY={3}>
-          {secondaryAction && responsiveMinWidth !== 'xs' && (
-            <UpsellAction type="secondary" data={secondaryAction} />
-          )}
-          {primaryAction && <UpsellAction type="primary" data={primaryAction} />}
-          {secondaryAction && responsiveMinWidth === 'xs' && (
-            <UpsellAction type="secondary" data={secondaryAction} stacked={!!secondaryAction} />
-          )}
-        </Box>
+        {children ? (
+          <Box smDisplay="flex" width="100%" justifyContent="end" smMarginEnd={4} smPaddingY={3}>
```

Comment:
@dangerismycat Updated per your offline advice but my TextField is still too short 🤔
<img width="785" alt="Screen Shot 2021-02-23 at 7 00 03 PM" src="https://user-images.githubusercontent.com/46180742/108949175-c7e68c80-7618-11eb-8609-577485066f7b.png">

Reply:
I believe you'll also need to set `width="100%"` on the `<form>` and then within UpsellForm, wrap `children` in `<Flex.Item flex="grow">`.

@christianvuerings is there a better/cleaner solution we're missing here?

Reply:
I'll be just dead code. `children` has priority over actions. We could throw errors but those only show on runtime. 

I remember the Flow issue we discussed (not exactly remember which component was and the case). Ideally yes, we would segment Upsell in 2, so when children are present we need to also include the submit button props. 

If we were to create `Upsell.primaryAction` &  `Upsell.secondaryAction`, this would put us in the same situation, how to communicate which props go with different use cases. 

Another brainstorm idea could be adding a prop variant = "form" | "actions" , and these props could help segment Flow.

I think for now doc + eslint work. At somepoint we'd have to discuss the official way of segmenting.